### ISSUE TITLE
Getting error Class 'PromiseTransform' incorrectly implements class 'Readable' and 'Writable'

### ISSUE THREAD

**Issue by shahabali1993** [signicode/scramjet#132]
I tried to install the latest scramjet version and getting this error on compilation.

**Node-Version**: 18.17.1 / **Scramjet-Version**: 4.36.9 

$ yarn compile
yarn run v1.22.19
$ tsc
node_modules/scramjet/.d.ts/index.d.ts:10:15 - error TS2720: Class 'PromiseTransform' incorrectly implements class 'Readable'. Did you mean to extend 'Readable' and inherit its members as a subclass?
  Type 'PromiseTransform' is missing the following properties from type 'ReadableBase': iterator, map, filter, forEach, and 11 more.

10 declare class PromiseTransform implements Readable, Writable {
                 ~~~~~~~~~~~~~~~~

node_modules/scramjet/.d.ts/index.d.ts:10:15 - error TS2720: Class 'PromiseTransform' incorrectly implements class 'Writable'. Did you mean to extend 'Writable' and inherit its members as a subclass?
  Property 'compose' is missing in type 'PromiseTransform' but required in type 'WritableBase'.

10 declare class PromiseTransform implements Readable, Writable {

**Comment by shahabali1993** [comment]
@MichalCz , @luccam-scr , @stokowski , @S4adam , can you please look into this issue?

**Comment by MichalCz** [comment]
Will check. Typescript definitions of `node/stream` used to be pretty bad, so if you could try to downgrade `@types/node` to something older (16) and rebuild then I'd know where we stand.

A small issue I'm having is the very outdated way the definitions are done - so there are two options: if method signatures need changing it will be problematic, if only missing props are added, this should be easy.

I'm also seeing a small issue with the tests, but that's a false positive.

Anyway @shahabali1993, please confirm if downgrading `@types/node` to `16.x` allows you to build Scramjet. If we're lucky that won't make your own build fail - if that's the case let me know, I'll try to set up a small test repo.

**Comment by shahabali1993** [comment]
@MichalCz thank you for the quick response, downgrading `@types/node` to 16.x worked with scramjet but now getting another issue with 'AbortSignal' related `@types/node` in file `globals.d.ts`.

Here's the link if you would like to follow:
https://github.com/DefinitelyTyped/DefinitelyTyped/issues/66617

**Scramjet** and **@types/node** will have to be upgraded to latest in the future though.

**Comment by MichalCz** [comment]
Ok. But that's a good lead - it is how I thought.

I'll follow up this week. If you can try to somehow mute this error in typescript config then please try to do so as a workaround. We'll fix this, but I can't give you dependable timelines yet (worst case scenario is end of the month).

M.

**Comment by shahabali1993** [comment]
Thanks @MichalCz , please do inform here when it is fixed.

**Comment by MichalCz** [comment]
Hey @shahabali1993 - I have the solution, but I still need to make this work with the build script. I'll inform you on the progress.

**Comment by MichalCz** [comment]
@shahabali1993 please check with `4.37.0`.

**Comment by shahabali1993** [comment]
@MichalCz I am unable to find 4.37.0, can you please specify where I can find it?
I tried updating the version in my package.json but it didn't work.

**Comment by MichalCz** [comment]
My bad, the package was published to the internal repo. Thanks for the heads up.

I'll release today

**Comment by MichalCz** [comment]
It's published now. :)

**Comment by shahabali1993** [comment]
@MichalCz thank you for the quick fix.

Verified `4.37.0`.
All errors have been resolved, `scram-jet 4.37.0` is now working with the latest version of `types/node 20.6.0`.

**Comment by MichalCz** [comment]
Fantastic! Another win for fixing at night, and another near-miss for deploying then. :D
